core/idrepo/rest-cxf/src/main/java/org/apache/syncope/core/rest/cxf/service/AbstractService.java

```diff
@@ -61,8 +61,23 @@ public abstract class AbstractService implements JAXRSService {
 
     protected String getActualKey(final AnyDAO<?> dao, final String pretendingKey) {
         String actualKey = pretendingKey;
-        if (!SyncopeConstants.UUID_PATTERN.matcher(pretendingKey).matches()) {
-            actualKey = dao.findKey(pretendingKey);
+        if (uriInfo.getPathParameters(true).containsKey("key")) {
```

Comment:
Can we add here `actualKey == null` as first condition the `if` clause? I'd rather avoid decoding path parameters if not strictly needed

Reply:
You would, if the path parameter has a key that does not match the supplied key in the request body. You could in theory be passing one key in the URL, and use another key in the body.

Reply:
what I was meaning is that if you call `getActualKey(dao, null)` the current proposed implemenation will not even look at path parameters, but just raise the exception `Key is null`

Reply:
This is how I'd refactor the current proposal:

```java
    protected String getActualKey(final AnyDAO<?> dao, final String pretendingKey) {
        String actualKey = pretendingKey;
        if (actualKey == null && uriInfo.getPathParameters(true).containsKey("key")) {
            final String keyInPath = uriInfo.getPathParameters(true).get("key").get(0);
            if (actualKey != null && !actualKey.equals(keyInPath)) {
                SyncopeClientException sce = SyncopeClientException.build(ClientExceptionType.InvalidRequest);
                sce.getElements().add("Key specified in request does not match key in the path");
                throw sce;
            }
        }
        if (actualKey == null) {
            SyncopeClientException sce = SyncopeClientException.build(ClientExceptionType.InvalidRequest);
            sce.getElements().add("Key is null");
            throw sce;
        }
        if (!SyncopeConstants.UUID_PATTERN.matcher(actualKey).matches()) {
            actualKey = dao.findKey(actualKey);
        }

        return actualKey;
    }
 ```

Reply:
you're right about the mismatch in my proposal; still, the current one will not even attempt to look at path parameters if the payload key is `null`

So it seems we are attempting to pursue to objectives here:
1. avoid NPE in case the payload bears a `null` key
2. if both path parameter and payload's  `key` are provided, check if they are the same